tests/testdata/programs/managed-ng-os/index.ts

```diff
@@ -306,8 +306,15 @@ const managedNodeGroupBottlerocketTaints = eks.createManagedNodeGroup("bottleroc
   },
 });
 
-// Create a simple AL2023 node group with a custom user data and a custom AMI
-const amiId = pulumi.interpolate`/aws/service/eks/optimized-ami/${cluster.core.cluster.version}/amazon-linux-2023/x86_64/standard/recommended/image_id`.apply(name =>
+// Fetch the previous minor version. This way we test that the user specified AMI is used instead of the default one.
+// MNGs tolerate a skew of 1 minor version
+export const customAmiId = pulumi.all([cluster.eksCluster.version]).apply(([version]) => {
+  const versionParts = version.split('.');
+  const majorVersion = parseInt(versionParts[0], 10);
+  const minorVersion = parseInt(versionParts[1], 10) - 1;
```

Comment:
This would break if k8s rolled out a new major version, but there's not indications this will happen in the next years and would make this computation unnecessarily complex.
If they ever do release one we can fix up the test (it'll break and not silently succeed).